app/src/test/java/org/mozilla/tv/firefox/pocket/PocketVideoFetchWorkerTest.kt

```diff
@@ -0,0 +1,90 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.pocket
+
+import androidx.work.ListenableWorker.Result
+import androidx.work.WorkerParameters
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.mozilla.tv.firefox.FirefoxApplication
+import org.mozilla.tv.firefox.utils.ServiceLocator
+
+class PocketVideoFetchWorkerTest {
+
+    private lateinit var worker: PocketVideoFetchWorker
+
+    @MockK private lateinit var context: FirefoxApplication
+    @Suppress("DEPRECATION") // We need endpoint raw until we move to the a-c impl.
+    @MockK private lateinit var endpointRaw: PocketEndpointRaw
+    @MockK private lateinit var store: PocketVideoStore
+
+    @MockK private lateinit var workerParams: WorkerParameters
+
+    @Before
+    fun setUp() {
+        MockKAnnotations.init(this)
+
+        every { context.applicationContext } returns context
+        every { context.serviceLocator } returns mockk<ServiceLocator>().also {
+            every { it.pocketEndpointRaw } returns endpointRaw
+            every { it.pocketVideoStore } returns store
+        }
+
+        worker = PocketVideoFetchWorker(context, workerParams)
+    }
+
+    @Test
+    fun `WHEN the endpoint fails to return a result THEN the job will retry`() {
+        coEvery { endpointRaw.getGlobalVideoRecommendations() } returns null
+        assertEquals(Result.retry(), worker.doWork())
+    }
+
+    @Test
+    fun `WHEN the endpoint returns a non-null result THEN the store decides what happens to the exact json string`() {
+        arrayOf(
+            "",
+            "{",
+            "{}",
+            "{\"json\": \"yeah!\"}"
+        ).forEachIndexed { i, json ->
+            coEvery { endpointRaw.getGlobalVideoRecommendations() } returns json
+            every { store.save(any()) } returns true
+
+            worker.doWork()
+
+            println("For index $i: $json")
```

Comment:
Remove this println?

Reply:
As discussed, this is intentional to make it clear where the test fails if it fails.